.github/workflows/publish-docker.yml

```diff
@@ -0,0 +1,51 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Build and Push Docker Image
+
+on:
+  push:
+    tags:
+      - 'apache-iceberg-[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v4
+      with:
+        distribution: zulu
+        java-version: 21
+    - name: Build Iceberg Open API project
+      run: ./gradlew :iceberg-open-api:shadowJar
+    - name: Login to Docker Hub
+      run: |
+        docker login -u ${{ secrets.DOCKERHUB_USER }} -p ${{ secrets.DOCKERHUB_TOKEN }}
+    - name: Build Docker Image
+      run: docker build -t apache/iceberg-rest-adapter -f docker/iceberg-rest-adapter-image/Dockerfile .
```

Comment:
What about using `apache/iceberg-rest-fixture` ? If we have a consensus, I can rename `apache/iceberg`.

Reply:
I am ok with the name `iceberg-rest-fixture`, it is a good name 👍

Reply:
Sounds good @jbonofre and @ajantha-bhat - I'm good with this new name suggestion as well

Reply:
Hi @jbonofre - the workflow is resulting in an `access denied` response at the proposed namespace:

```
Run docker push $DOCKER_REPOSITORY/$DOCKER_IMAGE_TAG:$DOCKER_IMAGE_VERSION
The push refers to repository [docker.io/apache/iceberg-rest-fixture]
0[10](https://github.com/apache/iceberg/actions/runs/12121028652/job/33791062874#step:8:11)919b3091e: Preparing
32bf0953c155: Preparing
88ad36680f9c: Preparing
32e16cb4dcee: Preparing
2573e0d81582: Preparing
denied: requested access to the resource is denied
```

Would you have the permission to verify access of the docker hub username to the proposed namespace `apache/iceberg-rest-fixture`?